Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -7,31 +12,201 @@ public static class BalanceStats
+            {
+                return "N/A.";
+            }
+
+            Dictionary<string, List<float>> peopleBalances = ArrayOfPeople(peopleAndBalances);
+            List<string> names = new List<string>();
+            var biggestBalanceEver = float.MinValue;
+
+            foreach (var personBalances in peopleBalances)
+            {
+                foreach (var balance in personBalances.Value)
+                {
+                    if (balance > biggestBalanceEver)
+                    {
+                        names.Clear();
+                        biggestBalanceEver = balance;
+                        names.Add(personBalances.Key);
+                    }
+                    if (Math.Round((decimal)balance, 2) == (Math.Round((decimal)biggestBalanceEver, 2)))
+                    {
+                        names.Add(personBalances.Key);
+                    }
+                }
+            }
+            return $"{BuildNamesString(names)} had the most money ever. ¤{biggestBalanceEver}.";
         }
 
         /// <summary>
         /// Return name and loss of a person with a biggest loss (balance change negative).
         /// </summary>
         public static string FindPersonWithBiggestLoss(string[] peopleAndBalances)
         {
-            return "";
-        }
+            if (peopleAndBalances == null || peopleAndBalances.Length == 0)
+            {
+                return "N/A.";
+            }
 
+            Dictionary<string, List<float>> peopleBalances = ArrayOfPeople(peopleAndBalances);
+            List<string> names = new List<string>();
+            var biggestLostEver = float.MaxValue;
+
+            foreach (var personBalances in peopleBalances)
+            {
+                for (int i = 0; i < personBalances.Value.Count - 1; i++)
+                {
+                    var balanceDifferences = personBalances.Value[i + 1] - personBalances.Value[i];
+                    if (balanceDifferences < biggestLostEver)
+                    {
+                        names.Clear();
+                        biggestLostEver = balanceDifferences;
+                        names.Add(personBalances.Key);
+                        continue;
+                    }
+                    if (Math.Round((decimal)balanceDifferences, 2) == (Math.Round((decimal)biggestLostEver, 2)))
+                    {
+                        names.Add(personBalances.Key);
+                    }
+                }
+            }
+
+            if (names.Count == 0)
+            {
+                return "N/A.";
+            }
+            return $"{BuildNamesString(names)} lost the most money. {BuildCurrancy(biggestLostEver)}.";
+        }
         /// <summary>
         /// Return name and current money of the richest person.
         /// </summary>
         public static string FindRichestPerson(string[] peopleAndBalances)
         {
-            return "";
+            if (peopleAndBalances == null || peopleAndBalances.Length == 0)
```

Comment:
Codacy found an issue: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5838869)